plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -262,8 +264,26 @@ func PrioritizeNodes(
 		go func(config algorithm.PriorityConfig) {
 			defer wg.Done()
 			weight := config.Weight
-			priorityFunc := config.Function
-			prioritizedList, err := priorityFunc(pod, nodeNameToInfo, nodes)
+
+			prioritizedList, err := func() (schedulerapi.HostPriorityList, error) {
+				if config.Function != nil {
+					return config.Function(pod, nodeNameToInfo, nodes)
+				}
+				prioritizedList := make(schedulerapi.HostPriorityList, 0, len(nodes))
+				for i := range nodes {
+					hostResult, err := config.Map(pod, meta, nodeNameToInfo[nodes[i].Name])
```

Comment:
At least not initially. Once everything is using this pattern, we will be in better position for optimizations and parallelization at the framework basis.